src/main/java/net/dv8tion/jda/core/requests/RateLimiter.java

```diff
@@ -74,10 +72,16 @@ public boolean isRateLimited(CompiledRoute route)
         }
     }
 
-    protected void shutdown()
+    protected void shutdown(long time, TimeUnit unit)
     {
         isShutdown = true;
 
+        pool.setKeepAliveTime(time, unit);
+        pool.allowCoreThreadTimeOut(true);
+    }
+
+    public void forceShutdown()
+    {
         pool.shutdownNow();
     }
```

Comment:
Because it is already set by `shutdown()` as`JDA#shutdownNow()` calls `JDA#shutdown()`